templates/connect-inject-deployment.yaml

```diff
@@ -60,19 +60,22 @@ spec:
                 {{ else if .Values.global.bootstrapACLs }}
                 -acl-auth-method="{{ template "consul.fullname" . }}-k8s-auth-method" \
                 {{- end }}
+                {{- if .Values.global.tls.enabled }}
+                -consul-ca-cert=/consul/tls/ca/tls.crt \
+                {{- end }}
                 {{- if .Values.connectInject.centralConfig.enabled }}
                 -enable-central-config=true \
                 {{- end }}
                 {{- if (and .Values.connectInject.centralConfig.enabled .Values.connectInject.centralConfig.defaultProtocol) }}
                 -default-protocol="{{ .Values.connectInject.centralConfig.defaultProtocol }}" \
                 {{- end }}
-{{- if .Values.connectInject.certs.secretName }}
+                {{- if .Values.connectInject.certs.secretName }}
```

Comment:
@ishustava does the generated envoy config file include the TLS context (cert/key)? I see that the CA is added inline.

Reply:
The envoy TLS context for the consul client (aka `local_agent`) doesn't need cert/key since it's not using mTLS client auth to talk to Consul. That's why it only has the CA cert. 

If your question is about authentication, we recommend using ACLs for production environments for that purpose, and when they are enabled in the Helm chart, Envoy will include an ACL token when talking to Consul.

Reply:
Thank you. I was more curious about mTLS for service to service (via envoy) communication.

Reply:
Ah, I see. mTLS for service to service is configured dynamically as you add and remove services by the Consul client agent. [These docs](https://www.consul.io/docs/connect/proxies/envoy#dynamic-configuration) talk a bit about that.

Reply:
Are you referring to the dynamically generated envoy-bootstrap.yaml file via the consul connect envoy -bootrap command? Is that where the cert/key should appear?

Reply:
No, that config is used only for bootstrapping envoy itself. Consul updates envoy config for services dynamically through the API, so you won't find them in a file. You could look at the config though through the envoy admin endpoint. All you need to do is this:

```
kubectl port-forward po/<your pod that has envoy sidecar container> 19000
```

Then go to the browser at http://localhost:19000. There you will see an option to dump the config. Alternatively, you could just curl the endpoint to get the config `curl http://localhost:19000/config_dump`

You could do other things through that endpoint too, like change log levels. Here are the [docs](https://www.envoyproxy.io/docs/envoy/latest/operations/admin) for the admin endpoint in case you'll find them useful.

Reply:
Fantastic! I appreciate your help. Thank you! This is awesome!